packages/events-example/src/deployments/pulumi.ts

```diff
@@ -0,0 +1,94 @@
+
+import { CliUx } from "@oclif/core";
+import { 
+  DestroyResult,
+    InlineProgramArgs, 
+    LocalWorkspace, 
+    LocalWorkspaceOptions, 
+    PulumiFn, 
+    Stack, 
+    StackSummary, 
+    UpResult
+} from "@pulumi/pulumi/automation";
+import { Instance } from "@theniledev/js";
+
+export interface PulumiFnGen {
+  (something: any): PulumiFn
+}
+
+export default class PulumiAwsDeployment {
+
+    projectName!: string;
+    private workspace!: LocalWorkspace;
+    private programGen!: PulumiFnGen;
+
+    static async create(projectName: string, opts: LocalWorkspaceOptions, program: PulumiFnGen): Promise<PulumiAwsDeployment> {
+        const ws = await LocalWorkspace.create(opts);
+        ws.installPlugin('aws', 'v4.0.0');
+        return new PulumiAwsDeployment(projectName, ws, program);
+    };
+
+    constructor(projectName: string, workspace: LocalWorkspace, programGenerator: PulumiFnGen) {
+        this.projectName = projectName;
+        this.workspace = workspace;
+        this.programGen = programGenerator;
+    }
+
+    async loadStacks(): Promise<{[key: string]: StackSummary}> {
+      const stacks = await (
+          await this.workspace.listStacks()
+        ).reduce(async (accP, stack) => {
+          const acc = await accP;
+          const fullStack = await this.getStack(stack.name, this.programGen({}));
+          const info = await fullStack.info();
+          if (info?.kind != 'destroy') {
+            acc[stack.name] = stack;
+            console.debug('adding stack', stack);
+          }
+          return acc;
+        }, Promise.resolve({} as { [key: string]: StackSummary }));
+      console.debug(stacks);
+      return stacks;
+    }
+
+    private async getStack(stackName: string, program: PulumiFn): Promise<Stack> {
+      const args: InlineProgramArgs = {
+        stackName,
+        projectName: this.projectName,
+        program,
+      };
+      const stack = await LocalWorkspace.createOrSelectStack(args);
+      await stack.setConfig('aws:region', { value: 'us-west-2' });    // TODO: generalize AWS config.
```

Comment:
would it be possible to do this TODO now?

Reply:
for sure.

Reply:
Done, though it's just the region for now. I think it at least gives something to talk about with customers.